src/core/crypto/aes_ecb.cpp

```diff
@@ -32,28 +32,37 @@
  */
 
 #include "aes_ecb.hpp"
+#include "common/error.hpp"
 
 namespace ot {
 namespace Crypto {
 
 AesEcb::AesEcb(void)
 {
-    mbedtls_aes_init(&mContext);
+    void *context = (void *)&Context;
+    Error err = otPlatCryptoAesInit(context);
```

Comment:
Should we add some assertions here?
Also, passing `void*` without the size seems quite unsafe. The platform code has no means to verify it has enough space to allocate its context. Not sure, what's the preferred way of solving it in OT (@abtink ?), but perhaps we should use some opaque data structure like `struct otAesContext { uint32_t rawData[SOME_CONSTANT]; };` and let the platform code to interpret it as the mbedtls or PSA context.

Reply:
I can create a opaque data structure, but that would mean each module will have a similar data structure defined by crypto module. The easiest solution would be to pass the context size along with the void pointer.

Reply:
Good point about assertions. I will update default mbedTLS implementation to return error as well.